Added serde serialization and deserialization as a feature

MarkJr94 opened the pull request:
Split commit off from PR #18 as recommended

MarkJr94 commented:
I rebased this, so it should be fine to merge now

alkis commented on src/dec128.rs:
```diff
@@ -76,6 +78,114 @@ impl Hash for d128 {
+        match self.state {
+            0 => {
+                self.state += 1;
+                Ok(Some(try!(serializer.serialize_struct_elt("bytes", &self.value.bytes))))
```
Should we be outputing bytes? In rustc_serialize I output the decimal representation. This is because it is lossless and easier to read. I think we should be consistent at the very least.

MarkJr94 replied:
I'll change it to the decimal representation, and I will add tests.

alkis commented:
Can we have a test for this? It can be the same as rustc-serialize.

MarkJr94 commented:
I added tests, and changed the serialization as recommended. I also fixed the cfg flags in mod `dec128::test` so that no combination of features ("serde ord_subset", "ord_subset", "ord_subset rustc-serialize", etc.) produces warnings.

alkis commented on src/dec128.rs:
```diff
@@ -76,6 +78,46 @@ impl Hash for d128 {
+    fn deserialize<D>(deserializer: &mut D) -> Result<d128, D::Error>
+        where D: Deserializer
+    {
+        let s = try!(deserializer.deserialize_str(d128Visitor));
```
Does this work? 

{ deserializer.deserialize_str(d128Visitor) work? }

(that is without try! or Ok(s))

MarkJr94 replied:
Sorry for delay, checking now.

MarkJr94 commented:
It should be good to merge now

alkis commented:
Thanks for playing!

MarkJr94 commented:
Thanks for accepting!

Also, could you bump the version in the Cargo.toml? the changes aren't breaking but maybe deserve it.

alkis commented:
Took a while but done. Thanks again for the contributions.
